src/imex.rs

```diff
@@ -595,20 +595,30 @@ async fn import_backup_old(context: &Context, backup_to_import: impl AsRef<Path>
 
     let mut all_files_extracted = true;
     for (processed_files_cnt, file_id) in file_ids.into_iter().enumerate() {
+        if context.shall_stop_ongoing().await {
+            all_files_extracted = false;
+            break;
+        }
+
         // Load a single blob into memory
-        let (file_name, file_blob) = context
+        match context
             .sql
             .query_row(
                 "SELECT file_name, file_content FROM backup_blobs WHERE id = ?",
                 paramsv![file_id],
                 |row| Ok((row.get::<_, String>(0)?, row.get::<_, Vec<u8>>(1)?)),
             )
-            .await?;
-
-        if context.shall_stop_ongoing().await {
-            all_files_extracted = false;
-            break;
+            .await
+        {
+            Ok((file_name, file_blob)) => {
```

Comment:
these file names are still parsed as utf-8.  i guess they only get written during the export and therefore never end up being from a C-core?